monitoring/nemo_global_mean_year_mean_timeseries.py

```diff
@@ -3,69 +3,81 @@
+        src = self.getarg("src", context)
+        dst = self.getarg("dst", context)
+        domain = self.getarg("domain", context)
+        varname = self.getarg("varname", context)
+        comment = (
+            f"Global average time series of **{varname}**. "
+            "Each data point represents the (spatial and temporal) "
+            "average over one leg."
+        )
         self.log_info(f"Create time series for ocean variable {varname} at {dst}.")
         self.log_debug(f"Domain: {domain}, Source file(s): {src}")
 
         self.check_file_extension(dst)
 
-        leg_cube = hlp.load_input_cube(src, varname)
+        leg_cube = helpers.load_input_cube(src, varname)
 
-        grid = self.getarg('grid', context, default='T')
+        grid = self.getarg("grid", context, default="T")
         with warnings.catch_warnings():
             # Suppress warning about insufficient metadata.
             warnings.filterwarnings(
-                'ignore',
+                "ignore",
                 "Collapsing a multi-dimensional coordinate.",
                 UserWarning,
-                )
+            )
             spatial_avg = leg_cube.collapsed(
-                ['latitude', 'longitude'],
+                helpers.spatial_coord_names(leg_cube),
                 iris.analysis.MEAN,
-                weights=hlp.compute_spatial_weights(domain, leg_cube.shape, grid=grid),
-                )
+                weights=helpers.spatial_weights(leg_cube, domain, grid),
+            )
         # Remove auxiliary time coordinate before collapsing cube
-        spatial_avg.remove_coord(spatial_avg.coord('time', dim_coords=False))
+        spatial_avg.remove_coord(spatial_avg.coord("time", dim_coords=False))
         ann_spatial_avg = spatial_avg.collapsed(
-            'time',
+            "time",
             iris.analysis.MEAN,
-            weights=hlp.compute_time_weights(spatial_avg),
+            weights=helpers.compute_time_weights(spatial_avg),
         )
         # Promote time from scalar to dimension coordinate
-        ann_spatial_avg = iris.util.new_axis(ann_spatial_avg, 'time')
+        ann_spatial_avg = iris.util.new_axis(ann_spatial_avg, "time")
 
-        ann_spatial_avg = hlp.set_metadata(
+        ann_spatial_avg = helpers.set_metadata(
             ann_spatial_avg,
-            title=f'{ann_spatial_avg.long_name} (Annual Mean)',
+            title=f"{ann_spatial_avg.long_name} (Annual Mean)",
             comment=comment,
-            )
+        )
 
         ann_spatial_avg.cell_methods = ()
         ann_spatial_avg.add_cell_method(
-            iris.coords.CellMethod('mean', coords='time', intervals='1 month')
-            )
-        ann_spatial_avg.add_cell_method(iris.coords.CellMethod('mean', coords='area'))
+            iris.coords.CellMethod("mean", coords="time", intervals="1 month")
+        )
+        ann_spatial_avg.add_cell_method(iris.coords.CellMethod("mean", coords="area"))
```

Comment:
This would have to be adjusted for 3D averages, probably? Not sure what has to be added though, "depth"?

something like
```python
if volume: 
    ann_spatial_avg.add_cell_method(iris.coords.CellMethod("mean", coords="depth"))
```